lambdas/handlers/bulk_upload_report_handler.py

```diff
@@ -0,0 +1,104 @@
+import csv
+import datetime
+import logging
+import os
+
+from boto3.dynamodb.conditions import Attr
+from botocore.exceptions import ClientError
+
+from services.dynamo_service import DynamoDBService
+from services.s3_service import S3Service
+from utils.decorators.ensure_env_var import ensure_environment_variables
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+@ensure_environment_variables(
+    names=[
+        "STAGING_STORE_BUCKET_NAME",
+        "BULK_UPLOAD_DYNAMODB_NAME",
+    ]
+)
+def lambda_handler(event, context):
+    db_service = DynamoDBService()
+    s3_service = S3Service()
+    try:
+        report_handler(db_service, s3_service)
+    except ClientError as e:
+        logger.error("Report creation failed")
+        logger.error(e.response)
+
+
+def report_handler(db_service, s3_service):
+    staging_bucket_name = os.getenv("STAGING_STORE_BUCKET_NAME")
+    start_time, end_time = get_times_for_scan()
+    report_data = get_dynamo_data(
+        db_service, int(start_time.timestamp()), int(end_time.timestamp())
+    )
+    if report_data:
+        file_name = f"Bulk upload report for {str(start_time)} to {str(end_time)}.csv"
+        write_items_to_csv(report_data, f"/tmp/{file_name}")
+    else:
+        file_name = f"Bulk upload report for {str(start_time)} to {str(end_time)}.txt"
+        write_empty_report(f"/tmp/{file_name}")
+    logger.info("Uploading new report file to S3")
+    s3_service.upload_file(
+        s3_bucket_name=staging_bucket_name,
+        file_key=f"reports/{file_name}",
+        file_name=f"/tmp/{file_name}",
+    )
+
+
+def get_dynamo_data(
```

Comment:
I don't like this name, it's not very descriptive.... Maybe 'discover_bulk_uploaded_documents' or something?
Petty comment I know, I'm sorry!

Reply:
Done